### DISCUSSION TITLE
Disable symbols / groups based on conditions

### DISCUSSION THREAD

**Discussion by rekup** [rspamd/rspamd#4792] [Q&A]
We are using an external DNS blocklist to check various parts of the message (urls, ip addresses etc.). This external provider charges based on the query volume. Thus, we are interested to minimize the amount of checks we perform to reduce costs.

We would like to perform the rbl checks only if a message does not already meet our `add_header` threshold. I suppose one way to achieve this would be to create a custom lua rule which checks the current score with something like `task:get_metric_result()` and then uses `task:disable_symbol(name)` if the score exceeds our threshold. We would then have to create a dependency for all rbl checks to our newly created rule to fix the evaluation order. Has anyone every tried something similar or a better idea how to disable symbols (or preferably groups) based on conditions?

Thanks!

**Comment by fatalbanana** [comment]
The way to dynamically disable some checks such that they would not be called at all is by using [settings](https://rspamd.com/doc/configuration/settings.html) but it's not appropriate for what you have in mind.

I'm afraid the only way I see something like this working is with a second call to Rspamd (or whichever other thing) which doesn't seem actually-recommendable and may be difficult to implement.
